src/core/runtime/dns-nodejs.js

```diff
@@ -3,19 +3,58 @@
 const dns = require('dns')
 
 module.exports = (domain, opts, callback) => {
-  dns.resolveTxt(domain, (err, records) => {
-    if (err) {
-      return callback(err, null)
-    }
+  resolveDnslink(domain)
+    .catch(err => {
+      // If the code is not ENOTFOUND then throw the error
+      if (err.code !== 'ENOTFOUND') throw err
```

Comment:
Errors in this resolution function come from one of two sources:
- `dns.resolveTxt` record throws an error if no TXT records are found, which has code `ENOTFOUND` (from node's dns module)
- There _are_ TXT records, but no dnslink records

The implementation here is simply to avoid having to check for 2 different error codes.

I can wrap the nodejs error by catching it, then throwing a differently constructed error if you'd like to switch to `ERR_NOT_FOUND` type of syntax, or the `err-code` module.

Alternatively I can use the `err-code` module in the dnslink propagated error and add a second conditional like `ERR_DNSLINK_NOT_FOUND`, to differentiate the error types.

Reply:
Gotcha! It's probably good practice for us to distinguish between the errors - I vote we throw `ERR_DNSLINK_NOT_FOUND` when there's no dnslink record and check for this here, as well as 'ENOTFOUND' from the `dns` module.

Reply:
Great, PR should be ready for review.